src/plugins/maven-workspace.ts

```diff
@@ -0,0 +1,345 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  WorkspacePlugin,
+  AllPackages,
+  DependencyGraph,
+  DependencyNode,
+  appendDependenciesSectionToChangelog,
+  addPath,
+} from './workspace';
+import {Version, VersionsMap} from '../version';
+import {CandidateReleasePullRequest} from '../manifest';
+import {PatchVersionUpdate} from '../versioning-strategy';
+import * as dom from '@xmldom/xmldom';
+import * as xpath from 'xpath';
+import {logger} from '../util/logger';
+import {dirname} from 'path';
+import {PomXml, parseDependencyNode} from '../updaters/java/pom-xml';
+import {Changelog} from '../updaters/changelog';
+import {ReleasePullRequest} from '../release-pull-request';
+import {PullRequestTitle} from '../util/pull-request-title';
+import {PullRequestBody} from '../util/pull-request-body';
+import {BranchName} from '../util/branch-name';
+
+interface Gav {
+  groupId: string;
+  artifactId: string;
+  version: string;
+}
+
+interface MavenArtifact extends Gav {
+  path: string;
+  name: string;
+  dependencies: Gav[];
+  testDependencies: Gav[];
+  pomContent: string;
+}
+
+const JAVA_RELEASE_TYPES = new Set(['java', 'java-bom', 'java-yoshi', 'maven']);
+const XPATH_PROJECT_GROUP =
+  '/*[local-name()="project"]/*[local-name()="groupId"]';
+const XPATH_PROJECT_ARTIFACT =
+  '/*[local-name()="project"]/*[local-name()="artifactId"]';
+const XPATH_PROJECT_VERSION =
+  '/*[local-name()="project"]/*[local-name()="version"]';
+const XPATH_PROJECT_DEPENDENCIES =
+  '/*[local-name()="project"]/*[local-name()="dependencies"]/*[local-name()="dependency"]';
+
+export class MavenWorkspace extends WorkspacePlugin<MavenArtifact> {
+  private async fetchPom(path: string): Promise<MavenArtifact | undefined> {
+    const content = await this.github.getFileContentsOnBranch(
+      path,
+      this.targetBranch
+    );
+    const document = new dom.DOMParser().parseFromString(content.parsedContent);
+
+    const groupNodes = xpath.select(XPATH_PROJECT_GROUP, document) as Node[];
+    if (groupNodes.length === 0) {
+      logger.warn(`Missing project.groupId in ${path}`);
+      return;
+    }
+    const artifactNodes = xpath.select(
+      XPATH_PROJECT_ARTIFACT,
+      document
+    ) as Node[];
+    if (artifactNodes.length === 0) {
+      logger.warn(`Missing project.artifactId in ${path}`);
+      return;
+    }
+    const versionNodes = xpath.select(
+      XPATH_PROJECT_VERSION,
+      document
+    ) as Node[];
+    if (versionNodes.length === 0) {
+      logger.warn(`Missing project.version in ${path}`);
+      return;
+    }
+    const dependencies: Gav[] = [];
+    const testDependencies: Gav[] = [];
+    for (const dependencyNode of xpath.select(
+      XPATH_PROJECT_DEPENDENCIES,
+      document
+    ) as Node[]) {
+      const parsedNode = parseDependencyNode(dependencyNode);
```

Comment:
Is the assumption that this list of dependencies represents local dependencies within a monorepo?

Reply:
This is all dependencies, but is only relevant when building the local dependency graph.